doc/nrf/drivers/uart_nrf_lpuart.rst

```diff
@@ -1,74 +1,98 @@
+The control protocol uses two additional lines alongside the standard TX and RX lines:
+
+* the *Request* line (REQ)
+* the *Ready* line (RDY)
+
+The REQ line of the first device is connected with the RDY line of the second device.
+It is configured as output and set to low.
+
+The RDY line is configured as input without pull up and the interrupt is configured to detect transition from low to high state.
 
-Low Power UART is built on top of the standard UART driver extended by control
-lines protocol. Because of that, configuration contains physical UART driver
-configuration which includes:
+Then:
 
-- Device Tree: TX and RX pin, baudrate, RTS and CTS pins must not be preset and
-hardware flow control must be disabled
-- Kconfig: Instance must have asynchronous API enabled with hardware bytes
-counting enabled. See :ref:`lpuart_sample` for configuration example.
+#. The transmitter initiate the transmission by calling :cpp:func:`uart_tx`.
+#. The driver reconfigures the REQ line to input with pull up and enables the detection of high to low transition.
+   The line is set to high, due to the pull up register, and that triggers an interrupt on the RDY line.
+#. On that interrupt, the driver starts the UART receiver.
+#. When the receiver is ready, the driver reconfigures the RDY line to output low.
+#. Then, the driver reconfigures the RDY line to input with pull up and enables the interrupt on detection of high to low transition.
+#. This sequence results in a short pulse on the line, as the line goes low and high.
+#. The initiator detects the pulse and starts the standard UART transfer.
+#. When the transfer is completed, the driver reconfigure the REQ line to the initial state: output set to low.
+   This results in a line state change.
+#. As the line goes low, the receiver detects the change.
+   This indicates that the UART receiver can be stopped.
 
-Low Power UART configuration includes:
-:option:`CONFIG_NRF_LPUART_UARTE_INSTANCE`: UART instance choice
-:option:`CONFIG_NRF_LPUART_REQ_PIN`: Request line pin
-:option:`CONFIG_NRF_LPUART_RDY_PIN`: Ready line pin
+Once the receiver acknowledges the transfer, it must be capable of receiving the whole transfer until the REQ line goes down.
+
+This requirement can be fullfilled in two ways:
+
+* By providing a buffer of the size of the maximum packet length
+* By continuously responding to the RX buffer request event.
+  The latency of the event handling must be taken into account in that case.
+  For example, a flash page erase on some devices may have a significant impact.
+
+Configuration
+*************
+
+The low power UART driver is built on top of the standard UART driver extended by control lines protocol.
+It is configured in the device tree as the child node of the UART instance which extends standard UART configuration with REQ and RDY lines.
+Additionally, the standard UART configuration must have flow control and flow control pins disabled.
+
+See the following example:
```

Comment:
Please state what this example shows.

Reply:
fixed, @nordic-krch please check this one